packages/astro/src/core/render-context.ts

```diff
@@ -163,6 +163,12 @@ export class RenderContext {
 							streaming,
 							this.routeData
 						);
+
+						if (result.cookies) {
+							for (const cookie of AstroCookies.consume(result.cookies)) {
+								response.headers.append('Set-Cookie', cookie);
```

Comment:
I don't think we should append the headers here. They are already appended further up in the response, here: https://github.com/withastro/astro/blob/cb4d07819f0dbdfd94bc4f084edf7720ada01323/packages/astro/src/core/app/index.ts#L346-L350

What I think is happening in this bug is that the `Response` has changed because of rewriting, so the AstroCookies object is not attached to it anymore. So what I think you can do is call [attachCookiesToResponse](https://github.com/withastro/astro/blob/main/packages/astro/src/core/cookies/response.ts#L5-L7) here and then when the cookies are consumed further up the chain they will be appended to the response there.

Reply:
Looks like they actually need to be merged, so I took a different approach